lib/inspec/objects/attribute.rb

```diff
@@ -71,5 +85,62 @@ def to_ruby
     def to_s
       "Attribute #{@name} with #{@value}"
     end
+
+    private
+
+    def validate_required(value)
+      # value will be set already if a secrets file was passed in
+      if (!@opts.key?(:default) && value.nil?) || (@opts[:default].nil? && value.nil?)
+        raise Inspec::AttributeRequiredError, "Attribute '#{@name}' is required and does not have a value."
+      end
+    end
+
+    def validate_type(type)
+      type = type.capitalize
+      abbreviations = {
+        'Num' => 'Numeric',
+        'Regex' => 'Regexp',
+      }
+      type = abbreviations[type] if abbreviations.key?(type)
+      raise Inspec::AttributeTypeError, "Type '#{type}' is not a valid attribute type." if !VALID_TYPES.include?(type)
+      type
+    end
+
+    def valid_numeric?(value)
+      Float(value)
+      true
+    rescue
+      false
+    end
+
+    def valid_regexp?(value)
+      # check for invalid regex syntex
+      Regexp.new(value)
+      true
+    rescue
+      false
+    end
+
+    def validate_value_type(value) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+      type = validate_type(@opts[:type])
+      return if type == 'Any'
+
+      invalid_type = false
+      if type == 'Regexp'
+        invalid_type = true if !value.is_a?(String) || !valid_regexp?(value)
+      elsif type == 'Numeric'
+        invalid_type = true if !valid_numeric?(value)
+      elsif type == 'Boolean'
+        invalid_type = true if ![true, false].include?(value)
+      elsif value.is_a?(Module.const_get(type)) == false
+        invalid_type = true
+      end
+
+      raise Inspec::AttributeValidationError, "Attribute '#{@name}' with value '#{value}' does not validate to type '#{type}'." if invalid_type == true
+    end
```

Comment:
You *must* re-enable rubocop checks after a block.  Rubocop variances are FILE-scoped, not block-scoped.  This causes much gnashing of teeth and rending of clothes; see https://github.com/rubocop-hq/rubocop/issues/1690#issuecomment-286347494